labS9J1/Sources/labo.c

```diff
@@ -1 +1,83 @@
-﻿#include "labo.h"
\ No newline at end of file
+﻿#include "labo.h"
+
+Node* create_node(void* data) {
+    Node* new_node = malloc(sizeof(Node));
+    if (!new_node) {
+        perror("Échec allocation Node");
+        exit(EXIT_FAILURE);
+    }
+
+    new_node->data = data;
+    memset(new_node->adj, 0, sizeof(new_node->adj));
+    new_node->visited = 0;
+    new_node->len = 0;
+
+    new_node->revPath = malloc(sizeof(QNode)); //Je n'ai pas compris l'erreur dans cette ligne
+    if (!new_node->revPath) {
+        perror("Échec allocation QNode");
+        free(new_node);
+        exit(EXIT_FAILURE);
+    }
+
+    new_node->revPath->node = new_node;
+    new_node->revPath->prev = NULL;
+
+    return new_node;
+}
+void add_adjacent_node(Node* root, Node* node) {
+    if (root->len < UINT8_MAX) {
+        root->adj[root->len++] = node;
+    }
+    else {
+        fprintf(stderr, "Erreur : trop de voisins pour le noeud %d\n", root->data);
+    }
+}
+int dfs(Node* root[], int len, Node* curr, void* key, Stack* s) {
+    if (curr == NULL) {
+        return 0; 
```

Comment:
Non, le noeud est NULL au début...:
```suggestion
        curr = root[0];
```